### DISCUSSION TITLE
Unable to use WeakReferenceMessenger in MAUI .NET 8

### DISCUSSION THREAD

**Discussion by IXChinmayD** [dotnet/maui#21795] [Upgrading from Xamarin]
I have used the concept of **MessagingCenter** extensively in the Xamarin Forms application to perform some actions based on the data change. Now I am migrating to MAUI .NET 8. I have a code snippet below where classes Page1, Page2, and Page3 can be **ContentPage** or **ContentView**. I am not able to implement the code below using the **WeakReferenceMessenger** concept.     

   

```

//In Page 2  
MessagingCenter.Subscribe<Page1>(this, "Trigger1", (sender) =>  
{  
     Method1();  
});  


 //In Page 2  
MessagingCenter.Subscribe<Page3>(this, "Trigger2", (sender) =>  
{  
    Method 2();  
});  


// In Page 1  
MessagingCenter.Send<Page1>(this, "Trigger1");  
   
// In Page 3  
MessagingCenter.Send<Page3>(this, "Trigger2");  

```
 

How to identify while subscribing only Method1() in Trigger1 should run if I have triggered a send request for Trigger1?  

Is there any sample project for the same for **WeakReferenceMessenger**?  

I am unable to Pass Trigger1 and Trigger2 messages in **WeakReferenceMessenger**. How can I achieve the same result using WeakReferenceMessenger?

**Comment by prabhavmehra** [comment]
You can use `WeakReferenceMessenger` similar to what you have right now. Bit confused on what you are asking here?

From my understanding,
You can just do `RegisterAll` to register all at once or `Register` with the type of message to register each message separately.
If you have 2 different messages for each method, Once you implement the `IRecipient` for each message and write the `Receive` method, you can be assured that Method1 and Method2 will run for each message respectively.

if you have the same message for both methods then you can add some sort of property/check to your class object.

**Reply by prabhavmehra** [reply]
The documentation should serve you well with the above issue ( if it is )

**Reply by IXChinmayD** [reply]
Thank you for your suggestion.

**Reply by prabhav-mehra** [reply]
I doubt having a check like this is helpful as it indicates that some piece of unexpected code is being triggered. Maybe a different solution for whatever workflow you are going for might be helpful.
